### PR TITLE
Feature/close mango account

### PR DESCRIPTION
**Pull request by abrzezinski94**

### CODE REVIEW

File: ts/client/src/client.ts [hunk 1]
```diff
@@ -829,6 +830,85 @@ export class MangoClient {
...
+  ): Promise<TransactionSignature> {
+    const instructions: TransactionInstruction[] = [];
+    const healthAccountsToExclude: PublicKey[] = [];
+
+    for (const serum3Account of mangoAccount.serum3Active()) {
+      const serum3Market = group.serum3MarketsMapByMarketIndex.get(
+        serum3Account.marketIndex,
+      )!;
+
+      const openOrders = mangoAccount.serum3.find(
```
Comment (microwavedcola1) [inline ts/client/src/client.ts:845]: you could just get oo account form serum3Account

File: ts/client/src/client.ts [hunk 2]
```diff
@@ -829,6 +830,85 @@ export class MangoClient {
...
   }
 
+  public async emptyAndCloseMangoAccount(
+    group: Group,
+    mangoAccount: MangoAccount,
+  ): Promise<TransactionSignature> {
+    const instructions: TransactionInstruction[] = [];
+    const healthAccountsToExclude: PublicKey[] = [];
+
+    for (const serum3Account of mangoAccount.serum3Active()) {
```
Comment (microwavedcola1) [inline ts/client/src/client.ts:840]: you want to cancel all serum3 orders also

File: ts/client/src/client.ts [hunk 3]
```diff
@@ -829,6 +830,85 @@ export class MangoClient {
...
+        group,
+        mangoAccount,
+        serum3Market.serumMarketExternal,
+      );
+      healthAccountsToExclude.push(openOrders);
+      instructions.push(closeOOIx);
+    }
+
+    for (const perp of mangoAccount.perpActive()) {
+      const perpMarketIndex = perp.marketIndex;
```
Comment (microwavedcola1) [inline ts/client/src/client.ts:859]: you want to cancel all perp open orders also

File: ts/client/src/client.ts (hunk 2 above)
Comment (abrzezinski94) [reply ts/client/src/client.ts:840]: We guide user to do it by them selfs like in v3 https://pbs.twimg.com/media/FJU_ub3XoA0AEdn.png

File: ts/client/src/client.ts (hunk 3 above)
Comment (abrzezinski94) [reply ts/client/src/client.ts:859]: user need to cancel them before closing https://pbs.twimg.com/media/FJU_ub3XoA0AEdn.png

**Review by microwavedcola1 — APPROVED** [review]

**Pull request merged**